### PR TITLE
[WJ-401] C FFI for ftml

### PR DESCRIPTION
**Pull request by emmiegit**
This PR adds an FFI for ftml. This C FFI (foreign function interface) allows any language which permits C API-compatible binding to run ftml functions within the same process. It only exposes the main rendering methods, to avoid excess FFI memory copying for each step. Consequently, it exposes parse warnings in the HTML or text output in addition to the regular fields.

This does not add any FFI interaction from PHP, it is purely for ftml itself.

### CODE REVIEW

File: ftml/README.md [hunk 1]
```diff
@@ -36,8 +36,16 @@ You can use this as a dependency by adding the following to your `Cargo.toml`:
 ftml = "0.4"
```
Comment (Monkatraz) [inline ftml/README.md:36]: Should probably update this in the readme, lol

**Review by Monkatraz — APPROVED** [review]
This looks like it wasn't fun to do.

File: ftml/README.md (hunk 1 above)
Comment (emmiegit) [reply ftml/README.md:36]: I haven't deployed any new versions of ftml to crates.io

Comment (Monkatraz) [reply ftml/README.md:36]: Oops, nevermind then.

**Comment by emmiegit** [comment]
thanks @Monkatraz

**Pull request merged**